controllers/tlspolicy_controller.go

```diff
@@ -196,28 +202,55 @@ func (r *TLSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return nil
 	}
 
+	ok, err = kuadrantgatewayapi.IsCertManagerInstalled(mgr.GetRESTMapper())
+	if err != nil {
+		return err
+	}
+	if !ok {
+		r.Logger().Info("TLSPolicy controller disabled. CertManager was not found")
+		return nil
+	}
+
 	gatewayEventMapper := mappers.NewGatewayEventMapper(mappers.WithLogger(r.Logger().WithName("gatewayEventMapper")), mappers.WithClient(mgr.GetClient()))
 
+	issuerStatusChangedPredicate := predicate.Funcs{
+		UpdateFunc: func(ev event.UpdateEvent) bool {
+			oldPolicy, ok := ev.ObjectOld.(certmanagerv1.GenericIssuer)
+			if !ok {
+				return false
+			}
+			newPolicy, ok := ev.ObjectNew.(certmanagerv1.GenericIssuer)
+			if !ok {
+				return false
+			}
+			oldStatus := oldPolicy.GetStatus()
+			newStatus := newPolicy.GetStatus()
+			return !reflect.DeepEqual(oldStatus, newStatus)
+		},
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.TLSPolicy{}).
+		Owns(&certmanagerv1.Certificate{}).
 		Watches(
 			&gatewayapiv1.Gateway{},
 			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, object client.Object) []reconcile.Request {
 				return gatewayEventMapper.MapToPolicy(ctx, object, &v1alpha1.TLSPolicy{})
 			}),
 		).
+		Watches(
+			&certmanagerv1.Issuer{},
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, object client.Object) []reconcile.Request {
+				return mapIssuerToPolicy(ctx, mgr.GetClient(), r.Logger(), object)
+			}),
+			builder.WithPredicates(issuerStatusChangedPredicate),
+		).
+		Watches(
+			&certmanagerv1.ClusterIssuer{},
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, object client.Object) []reconcile.Request {
+				return mapClusterIssuerToPolicy(ctx, mgr.GetClient(), r.Logger(), object)
+			}),
+			builder.WithPredicates(issuerStatusChangedPredicate),
+		).
```

Comment:
Will these not panic if cert manager CRDs are missing?

Reply:
True, I haven't explicitly checked, but I can add these CRDs to the `IsCertManagerInstalled` function once I have

Reply:
Updated to account for `Issuer` & `ClusterIssuer` CRDs